### DISCUSSION TITLE
CRITICAL - Fetch code no longer work, impacting dozens of users (seems resolved)

### DISCUSSION THREAD

**Discussion by AlexanderCryptoOwl** [supabase/supabase#5801] [Questions] [answered]
EDIT: I think was solved now by changing .eq to .or like the other code. What is strange why the error wasn't thrown the last 4-5 days but only now.

Hi, I have a cloud function that I created a few days ago and it worked fine until today when suddenly Supabase is throwing error despite nothing changed on my side.

This is part of the code where the error is happening

```
var ts = new Date();
ts.setHours(ts.getHours() - 168) // 7*24hours

// get custom alerts for subscribers or free trials
const getCustomAlerts = async() => {
    let { data, error } = await supabase
    .from('custom_alerts')
    .select('*, profiles!inner(email)')
    .eq(`created_at.gt.${ts.toISOString()},is_subscribed.eq.true`, { foreignTable: "profiles" })

    if(error) {
        console.error(error)
    }
    console.log(data) 
}
```

Supabase Error

```
{
  message: "Cannot apply filter because 'created_at' is not an embedded resource in this request",
  hint: "Verify that 'created_at' is included in the 'select' query parameter."
}
```

It's very strange cause I have a similar fetch request in another function which still works fine that looks like this 

```
var ts = new Date();
ts.setHours(ts.getHours() - 168) // 7*24hours

const supabaseNotifications = async() => {
    let { data, error } = await supabase
    .from('notifications')
    .select('*, crypto_signals!inner(coin, coin_abr, combo_change, combo_signal, combo_prev_signal), profiles!inner(email)')
    .eq('crypto_signals.combo_change', true)
    .or(`created_at.gt.${ts.toISOString()},is_subscribed.eq.true`, { foreignTable: "profiles" })

    if(error) {
        console.error(error)
        return
    }
    return data
}
```

**Comment by steve-chavez** [comment, marked as answer]
Hello,

>     .eq(`created_at.gt.${ts.toISOString()},is_subscribed.eq.true`, { foreignTable: "profiles" })

The above line is incorrect, it's using an [or filter](https://supabase.com/docs/reference/javascript/or) syntax inside an [eq filter](https://supabase.com/docs/reference/javascript/eq), basically there's no condition being applied there.

Previous versions of PostgREST were lenient with this kind of errors but some users where doing delete/update(on tables without RLS) requests with incorrect/ignored filters resulting in a large amount of rows being affected, so PostgREST is now strict with ignored filters(
https://github.com/PostgREST/postgrest/issues/1771) and it errs accordingly.

> EDIT: I think was solved now by changing .eq to .or like the other code. What is strange why the error wasn't thrown the last 4-5 days but only now didn't allow me.

Yes, correcting the filter is the appropriate solution, we recently upgraded all instances, so that's why you didn't see the error before.

**Reply by steve-chavez** [reply]
Sorry about the inconveniences caused, we're planning to be more explicit with our services versioning and put a time window for upgrades so you can prevent breaking changes.

**Reply by AlexanderCryptoOwl** [reply]
Hi thanks for the quick reply, just happy the function was a cron job :D. I should have seen it in the code beforehand, cheers.
